### PR TITLE
Fix block templates not being rendered in extension taxonomies

### LINKED ISSUE #44453: Shop Page Filter Query Causes Inconsistent Block UI Rendering Without Product Attribute Lookup Table
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When including a filter query on the Shop page, filtering products with the catalog table's product lookup table results in the Block UI rendering correctly. However, if not using the product lookup table, the Block UI fails to render properly. This issue arises because the get_template_loader_default_file function determines there is no taxonomy_has_block_template and thus renders taxonomy-product-attribute.php. Notably, this problem does not occur with Storefront since it is not a block theme.

https://github.com/woocommerce/woocommerce/assets/10369528/9c07a588-3de8-44b0-819e-db021a915d5a

### Expected behavior

The variation in template rendering based on the use of the attribute table is unusual and not desirable. When using filters on the Shop page, utilizing the block renderer seems to be the correct approach regardless of the attribute table's usage status.

### Actual behavior

When not using the catalog table's product lookup table, the Block UI does not render correctly.

### Steps to reproduce

1. Enable the use of the product lookup table in the WordPress admin page.
2. Navigate to WooCommerce > Shop page.
3. Attempt product filtering using AJAX. (Directly entering a URL with the filter query string is also viable.)
4. Observe normal operation.
5. Disable the use of the product lookup table in the WordPress admin page.
6. Confirm the issue where the Block UI fails to render properly when not using the product lookup table.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): http://localhost:8000
Site address (URL): http://localhost:8000
WC Version: 8.4.0
REST API Version: ✔ 8.4.0
WC Blocks Version: ✔ 11.6.2
Action Scheduler Version: ✔ 3.7.0
Log Directory Writable: ✔
WP Version: ❌ 6.3.2 - 새 버전의 워드프레스를 사용할 수 있습니다(6.4.3).
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: ko_KR
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.57 (Debian)
PHP Version: 8.1.25
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.88.1
OpenSSL/3.0.11

SUHOSIN Installed: –
MySQL Version: 5.7.44
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ 귀하의 서버는 SoapClient 클래스가 활성화되어있지 않습니다 - 몇 결제 게이트웨이는 SOAP를 사용하며
올바르게 동작하지 않을 수 있습니다.

DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.4.0
WC Database Prefix: wp_
총 데이터베이스 크기: 93.95MB
데이터베이스 데이터 크기: 81.22MB
데이터베이스 인덱스 크기: 12.73MB
wp_woocommerce_sessions: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: 데이터: 0.02MB + 색인: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: 데이터: 0.02MB + 색인: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: 데이터: 0.02MB + 색인: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: 데이터: 0.02MB + 색인: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_woocommerce_log: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: 데이터: 0.05MB + 색인: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_commentmeta: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_comments: 데이터: 0.02MB + 색인: 0.09MB + Engine InnoDB
wp_links: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_options: 데이터: 4.05MB + 색인: 0.06MB + Engine InnoDB
wp_postmeta: 데이터: 67.59MB + 색인: 4.03MB + Engine InnoDB
wp_posts: 데이터: 8.52MB + 색인: 7.06MB + Engine InnoDB
wp_snippets: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_termmeta: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_terms: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_term_relationships: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_term_taxonomy: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_usermeta: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_users: 데이터: 0.02MB + 색인: 0.05MB + Engine InnoDB
wp_wc_admin_notes: 데이터: 0.06MB + 색인: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: 데이터: 0.05MB + 색인: 0.02MB + Engine InnoDB
wp_wc_category_lookup: 데이터: 0.02MB + 색인: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_wc_download_log: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_wc_orders: 데이터: 0.02MB + 색인: 0.11MB + Engine InnoDB
wp_wc_orders_meta: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_wc_order_addresses: 데이터: 0.02MB + 색인: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: 데이터: 0.02MB + 색인: 0.06MB + Engine InnoDB
wp_wc_order_stats: 데이터: 0.02MB + 색인: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: 데이터: 0.02MB + 색인: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: 데이터: 0.02MB + 색인: 0.09MB + Engine InnoDB
wp_wc_rate_limits: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: 데이터: 0.02MB + 색인: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB
wp_wc_webhooks: 데이터: 0.02MB + 색인: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 29667
page: 8
post: 2
product: 27
revision: 2
shop_order_placehold: 1
wp_global_styles: 1
wp_navigation: 1
wp_template: 3
wpc_product_badge: 2
yith_wcan_preset: 1
yith-wcbm-badge: 1

### Security ###

Secure connection (HTTPS): ❌
					상점에서 HTTPS를 사용하고 있지 않습니다. HTTPS 및 SSL 인증서에 대해 더 알아보기 .
Hide errors from visitors: ✔

### Active Plugins (4) ###

Code Snippets: Code Snippets Pro – 3.6.2
우커머스: 오토매틱스 – 8.4.0(8.5.2 버전으로 업데이트할 수 있음)
WPC Badge Management for WooCommerce: WPClever – 2.3.7
YITH WooCommerce Ajax Product Filter: YITH – 4.30.0

### Inactive Plugins (4) ###

Akismet Anti-spam: Spam Protection: Automattic - Anti-spam Team – 5.3
Category Order and Taxonomy Terms Order: Nsp-Code – 1.8
Hello Dolly: Matt Mullenweg – 1.7.2
YITH WooCommerce Badge Management: YITH – 3.2.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: KRW (₩)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### WC Pages ###

상점 기준 경로: #6 - /shop/
장바구니: #7 - /cart/
결제: #8 - /checkout/
내 계정: #9 - /my-account/
이용약관: ❌ 페이지가 설정되지 않았습니다

### Theme ###

Name: Blockskit Shop
Version: 1.0.0
Author URL: https://blockskit.com
Child Theme: ❌ – 직접 제작한 테마가 아닌 경우 부모 테마의 우커머스 코드를 수정하는 것보다 하위 테마를 사용할 것을 권장합니다. 다음을 참고하세요: 하위 테마 생성하기
WooCommerce Support: ❌ 선언되지 않음

### Templates ###

Overrides: /var/www/html/wp-content/plugins/woocommerce/packages/woocommerce-blocks/templates/notices/error.php
/var/www/html/wp-content/plugins/woocommerce/packages/woocommerce-blocks/templates/notices/notice.php
/var/www/html/wp-content/plugins/woocommerce/packages/woocommerce-blocks/templates/notices/success.php

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-external-affiliate
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2023-12-13 06:44:23 +09:00
Options: ✔
Notes: 65
Onboarding: completed

### Action Scheduler ###

대기 중: 74
Oldest: 2023-12-13 15:44:26 +0900
Newest: 2024-02-08 20:22:50 +0900

### Status report information ###

Generated at: 2024-02-08 21:29:00 +09:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by Aljullu**
### Changes proposed in this Pull Request:

Alternative approach from #44646.

Closes https://github.com/woocommerce/woocommerce/issues/44453. (**Update:** it turns out https://github.com/woocommerce/woocommerce/pull/44537 already fixed this issue)

In some circumstances, `woocommerce_has_block_template` wasn't set to `true` correctly, that meant that we were rendering a broken PHP template instead of the correct block template.

The issue was only reproducible in block themes with certain custom WooCommerce templates (ie: Tsubaki, which has the `archive-product.html` template), and were only reproducible in these cases:

* When rendering a product archive page belonging to a taxonomy with no template in the theme. That's excluding the category, tag and attribute taxonomies. So it was only reproducible in taxonomies added by extensions.
* ~When rendering the Shop page with certain filters (what's described in #44453).~ (**Update:** This was already fixed in `trunk` thanks to https://github.com/woocommerce/woocommerce/issues/44453)

This PR removes a check we had in the code that would only set `woocommerce_has_block_template` if the template was not in the theme.

### How to test the changes in this Pull Request:

**Verify correct template is rendered with certain settings (#44453)**

(**Update**: this issue was actually fixed in https://github.com/woocommerce/woocommerce/pull/44537, but I'm adding the testing steps here just to verify there are no regressions)

1. With a theme with custom WooCommerce templates (ie: Tsubaki).
2. Go to WooCommerce > Settings > Products > Advanced and disable _Use the product attributes lookup table for catalog filtering_.
3. With a block theme with custom WooCommerce templates (ie: Tsubaki) go to `/shop/?product_cat=Clothing&filter_size=large`. Note: the values of `product_cat` and `filter_xyz` might vary depending on the categories and attributes of your store. In case of doubt, you can import the [products from `sample-data`](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/sample-data).
4. Verify the page is rendered correctly. Ie: verify the header is displayed, verify the page has the _Shop_ title, verify the styles are loaded, etc.

Before | After
--- | ---
image | image

**Verify taxonomies added by extensions are rendered correctly (testing for devs)**

1. With a theme with custom WooCommerce templates (ie: Tsubaki).
2. Install WooCommerce Brands.
3. In the admin, create a brand, edit a product and assign that product to the brand you just created.
4. Visit the brand page in the frontend (the page should render correctly).
5. Edit `class-wc-brands-block-templates.php` and comment out this line: `add_filter( 'woocommerce_has_block_template', array( $this, 'has_block_template' ), 10, 2 );`.
6. Visit the brand page again. Verify the page still renders correctly.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Blocks/Templates/CartTemplate.php
Comment (imanish003) [inline plugins/woocommerce/src/Blocks/Templates/CartTemplate.php:1]: Nit: I am seeing multiple PHPCS warnings in this file. Can you see those, too? 

image
image

File: plugins/woocommerce/src/Blocks/Templates/CheckoutTemplate.php
Comment (imanish003) [inline plugins/woocommerce/src/Blocks/Templates/CheckoutTemplate.php:1]: Nit: I am seeing PHPCS warning in this file. Can you see it, too?

image

File: plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php
Comment (imanish003) [inline plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php:1]: Nit: I am seeing PHPCS warning in this file. Can you see it, too?

image

File: plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering
Comment (imanish003) [inline plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: I don't think it's related to this PR, but "Subscribe to newsletter" isn't rendering correctly in Footer with Tsubaki theme, as you can see in screenshot below. Are you able to reproduce it? 

image

Comment (imanish003) [inline plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: I encountered this error with my local setup, so I created a new site to test the PR. I'm mentioning this here in case it affects any existing sites.
image

**Review by imanish003 — CHANGES_REQUESTED** [review]
Hey @Aljullu,

I've tested the changes and everything is functioning as anticipated. I've left some minor comments and will await your response before giving my approval to this PR 🚀.

Thank you for your excellent work around templates so far 🙌🏻

File: plugins/woocommerce/src/Blocks/Templates/CartTemplate.php
Comment (WunderBart) [reply plugins/woocommerce/src/Blocks/Templates/CartTemplate.php:1]: I'm having some issues running `phpcs` in the monorepo, but shouldn't the CI also catch linting errors? It looks like all the checks are passing. 🤔

Comment (imanish003) [reply plugins/woocommerce/src/Blocks/Templates/CartTemplate.php:1]: @WunderBart Were you able to reproduce the PHPCS error? Maybe I have misconfigured my PHPCS extension in VSCode 😅

Comment (imanish003) [reply plugins/woocommerce/src/Blocks/Templates/CartTemplate.php:1]: > but shouldn't the CI also catch linting errors? It looks like all the checks are passing. 🤔

It should, maybe @ObliviousHarmony can provide more info on this? 👀

Comment (ObliviousHarmony) [reply plugins/woocommerce/src/Blocks/Templates/CartTemplate.php:1]: WooCommerce uses `phpcs-changed` to lint files instead of `phpcs`. This limits linting to lines that have changed in files that have changed. This was done because WooCommerce violates a number of its own linting rules. Rather than attempting to fix all of these, a policy of fixing them as lines were touched was adopted.

Comment (WunderBart) [reply plugins/woocommerce/src/Blocks/Templates/CartTemplate.php:1]: I checked by running linter against the file but cannot see the same errors:

```sh
$ pnpm lint:php src/Blocks/Templates/CartTemplate.php

> @woocommerce/plugin-woocommerce@8.8.0 lint:php /Users/bart/Code/woocommerce/woocommerce/plugins/woocommerce
> composer run-script phpcs "src/Blocks/Templates/CartTemplate.php"

> phpcs -s -p 'src/Blocks/Templates/CartTemplate.php'
. 1 / 1 (100%)


Time: 139ms; Memory: 14MB
```

File: plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php
Comment (WunderBart) [reply plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php:1]: I can see the error only when runnning `phpcs` directly against that file (not caught by `phpcs-changed`):

```
$ ./vendor/bin/phpcs --standard=WooCommerce-Core src/Blocks/Templates/SingleProductTemplate.php

--------------------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------------------------------------
  1 | ERROR | [ ] Filenames should be all lowercase with hyphens as word separators. Expected
    |       |     singleproducttemplate.php, but found SingleProductTemplate.php.
  1 | ERROR | [ ] Class file names should be based on the class name with "class-" prepended. Expected
    |       |     class-singleproducttemplate.php, but found SingleProductTemplate.php.
  1 | ERROR | [ ] Missing file doc comment
 54 | ERROR | [x] Short array syntax is not allowed
--------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------
```

File: plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering
Comment (WunderBart) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: I can confirm this is true for `trunk` as well.

**Review by WunderBart — COMMENTED** [review]
~I can repro https://github.com/woocommerce/woocommerce/issues/44453, but this PR doesn't seem to fix it. Here's what I see for the following themes on both `trunk` and this branch:~

| 2024 | Tsubaki | Storefront |
| ------------- | ------------- | ------------- |
| Screenshot 2024-03-13 at 15 46 05 | Screenshot 2024-03-13 at 15 45 15 | Screenshot 2024-03-13 at 15 45 36 |

~Looks like Storefront is affected as well, and only 2024 seems to render the results correctly.~

**EDIT**

I misunderstood the issue, but it seems to have been actually fixed on `trunk` as well as the page does not render differently on this branch.

File: plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering
Comment (WunderBart) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: I tried to reproduce with a fresh site but I can't see these errors.

Comment (WunderBart) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: @Aljullu do you know if this is a known issue?

File: plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php
Comment (WunderBart) [reply plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php:1]: > not caught by phpcs-changed

By that, I meant that it seems to be aligned with the current linting standards! 😄

File: plugins/woocommerce/src/Blocks/Templates/CheckoutTemplate.php
Comment (WunderBart) [reply plugins/woocommerce/src/Blocks/Templates/CheckoutTemplate.php:1]: As above, linter does not show any errors:

```sh
$ pnpm lint:php src/Blocks/Templates/CheckoutTemplate.php

> @woocommerce/plugin-woocommerce@8.8.0 lint:php /Users/bart/Code/woocommerce/woocommerce/plugins/woocommerce
> composer run-script phpcs "src/Blocks/Templates/CheckoutTemplate.php"

> phpcs -s -p 'src/Blocks/Templates/CheckoutTemplate.php'
. 1 / 1 (100%)


Time: 145ms; Memory: 14MB
```

File: plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php
Comment (WunderBart) [reply plugins/woocommerce/src/Blocks/Templates/SingleProductTemplate.php:1]: Ditto, linter is clear:

```sh
$ pnpm lint:php src/Blocks/Templates/SingleProductTemplate.php

> @woocommerce/plugin-woocommerce@8.8.0 lint:php /Users/bart/Code/woocommerce/woocommerce/plugins/woocommerce
> composer run-script phpcs "src/Blocks/Templates/SingleProductTemplate.php"

> phpcs -s -p 'src/Blocks/Templates/SingleProductTemplate.php'
. 1 / 1 (100%)


Time: 167ms; Memory: 12MB
```

**Comment by Aljullu** [comment]
> I misunderstood the issue, but it seems to have been actually fixed on `trunk` as well as the page does not render differently on this branch.  

You're right, @WunderBart! It looks like the refactor PR (https://github.com/woocommerce/woocommerce/pull/44537) already fixed that issue. :smile: I updated the description of this PR and the refactor PR to account for that.

File: plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering
Comment (Aljullu) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: @imanish003 did you switch themes after saving a customization on the template? If so, I think you are experimenting the issue described in https://github.com/woocommerce/woocommerce/issues/42181, which is definitely something that we want to fix, but IMO not in the scope of this PR.

Comment (Aljullu) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: Looking at the source code of Tsubaki, I think it's intentional to have this shortcode in the footer. Tsubaki is not distributed on wordpress.org and is intended to be used on sites with Jetpack installed, so I'm pretty sure that's expected.

**Comment by Aljullu** [comment]
Thanks for the reviews, folks! I think all comments have been answered and I updated the PR description and testing steps to account for the fact that https://github.com/woocommerce/woocommerce/issues/44453 was already fixed in `trunk` (sorry for the confusion!). This PR is ready for another look.

PS: even though #44453 has been fixed in `trunk`, I still think there is value in merging this PR because (1) it simplifies the code and (2) solves the issue described under the _Verify taxonomies added by extensions are rendered correctly_ testing section.

File: plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering
Comment (imanish003) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: @Aljullu Yeah, I think you are right. I did customization before switching the themes. I faced the same issue in the past too & now I know that it's a known issue 🙌🏻

Comment (imanish003) [reply plugins/woocommerce/changelog/fix-44453-fix-block-templates-not-rendering:1]: @Aljullu That makes sense. Thanks for sharing the reason behind it 🙌🏻

**Review by imanish003 — APPROVED** [review]
I verified the changes again & everything is working as expected :shipit:

**Pull request merged**
